crates/adventure-simulator-client/src/main.rs

```diff
@@ -1,49 +1,350 @@
+        Vec3::new(-5.0, 0.5, -5.0),
+        Color::srgb(0.8, 0.8, 0.4),
+    );
+    spawn_prop(
+        &mut commands,
+        &mut meshes,
+        &mut materials,
+        Vec3::new(10.0, 0.75, 0.0),
+        Color::srgb(0.6, 0.3, 0.6),
+    );
+    spawn_prop(
+        &mut commands,
+        &mut meshes,
+        &mut materials,
+        Vec3::new(-10.0, 0.75, 0.0),
+        Color::srgb(0.3, 0.6, 0.6),
+    );
+
+    // Lighting
+    commands.spawn((
+        DirectionalLight {
+            illuminance: 10000.0,
+            shadows_enabled: true,
+            ..default()
+        },
+        Transform::from_xyz(10.0, 20.0, 10.0).looking_at(Vec3::ZERO, Vec3::Y),
+    ));
+
+    commands.insert_resource(AmbientLight {
+        color: Color::srgb(0.4, 0.4, 0.6),
+        brightness: 500.0,
+        affects_lightmapped_meshes: true,
+    });
+
+    // Camera
+    commands.spawn((
+        MainCamera,
+        Camera3d::default(),
+        Transform::from_xyz(0.0, CAMERA_HEIGHT, CAMERA_DISTANCE).looking_at(Vec3::ZERO, Vec3::Y),
+    ));
+
+    // UI - Controls text
+    commands.spawn((
+        ControlsText,
+        Text::new("WASD to move | Arrow keys to rotate camera"),
+        TextFont {
+            font_size: 20.0,
+            ..default()
+        },
+        TextColor(Color::WHITE),
+        Node {
+            position_type: PositionType::Absolute,
+            bottom: Val::Px(20.0),
+            left: Val::Px(20.0),
+            ..default()
+        },
+    ));
 
+    // Position indicator
     commands.spawn((
-        AdventureSimulatorClient {
-            id,
-            server_addr,
+        Text::new("Position: (0.0, 0.0)"),
+        TextFont {
+            font_size: 16.0,
+            ..default()
+        },
+        TextColor(Color::srgba(1.0, 1.0, 1.0, 0.7)),
+        Node {
+            position_type: PositionType::Absolute,
+            top: Val::Px(20.0),
+            left: Val::Px(20.0),
             ..default()
         },
-        InputTimeline(Timeline::from(
-            Input::default().with_input_delay(InputDelayConfig::fixed_input_delay(0)),
-        )),
     ));
+}
+
+fn spawn_prop(
+    commands: &mut Commands,
+    meshes: &mut ResMut<Assets<Mesh>>,
+    materials: &mut ResMut<Assets<StandardMaterial>>,
+    position: Vec3,
+    color: Color,
+) {
+    commands.spawn((
+        Mesh3d(meshes.add(Cuboid::new(1.0, position.y * 2.0, 1.0))),
+        MeshMaterial3d(materials.add(StandardMaterial {
+            base_color: color,
+            metallic: 0.2,
+            perceptual_roughness: 0.7,
+            ..default()
+        })),
+        Transform::from_translation(position),
+    ));
+}
+
+fn player_movement(
+    keyboard: Res<ButtonInput<KeyCode>>,
+    time: Res<Time>,
+    mut query: Query<&mut Transform, With<Player>>,
+    camera_query: Query<&Transform, (With<MainCamera>, Without<Player>)>,
+) {
+    let Ok(camera_transform) = camera_query.single() else {
+        return;
+    };
```

Comment:
```suggestion
    camera: Single<&Transform, (With<MainCamera>, Without<Player>)>,
) {
    let camera_transform = camera.into_inner();
```
Let's do `Single` so it emit errors/panics automatically